src/wasm-lib/kcl/src/execution/mod.rs

```diff
@@ -198,6 +198,14 @@ impl ExecState {
 
         Ok(id)
     }
+
+    pub fn length_unit(&self) -> UnitLen {
+        self.mod_local.settings.default_length_units
```

Comment:
Can we make a new type for these, basically, I think we don't want people to put _all_ the settings that exist today in the file header, it would be a subset (which is units only today) but could grow, of things people would want checked into git. So we seperate out appearance shit, from the shit that matters if that makes sense. For backwards compat we can keep the old struct w units for now, as long as there is a path forward with a new one and we can nuke the field in the old after, if that makes sense

Reply:
We still use the old per-project settings as the default if there are no per-file settings (I think the final step in the migration is deciding what to do here, whether we want to keep this behaviour or remove the per-project settings, but that requires a lot of UI changes so I haven't done it here).

There is a separate settings struct for the per-file stuff called `MetaSettings` which is in the ExecState (c.f., settings in the ExecContext for stuff specified outside the file).

Reply:
gotcha gotcha okay yeah i was just mentioning to frank youll likely need some help w the front-end bits here since i think it touches a lot

Reply:
MetaSettings is great, sorry i was looking for whatever that was but likely it was in another pr